src/MessagePack.UnityClient/Assets/Scripts/Tests/ShareTests/MessagePackSerializerTest.cs

```diff
@@ -33,21 +31,25 @@ public void NonGeneric()
             var writer = new MessagePackWriter(writerBytes);
 
             var data1 = MessagePackSerializer.Deserialize(t, MessagePackSerializer.Serialize(t, data)) as FirstSimpleData;
-            var data2 = MessagePackSerializer.Deserialize(t, MessagePackSerializer.Serialize(t, data, StandardResolver.Options)) as FirstSimpleData;
+            var data2 = MessagePackSerializer.Deserialize(t, MessagePackSerializer.Serialize(t, data, MessagePackSerializer.DefaultOptions)) as FirstSimpleData;
```

Comment:
Since `MessagePackSerializer.DefaultOptions` is mutable, this makes the test run potentially fail or test the wrong thing if this test runs after or at the same time as another test that changes this setting. Why is this change necessary?

Reply:
for Unity IL2CPP test, load generated resolve in assembly load.
https://github.com/neuecc/MessagePack-CSharp/blob/master/src/MessagePack.UnityClient/Assets/Scripts/Tests/Loader.cs

If set individually, #if ENABLE_IL2CPP is required in each test.
It's just too complicated so better to use DefaultOptions.

By the way, some of the tests for IL2CPP are failing.
This PR doesn't fix them all.
The reason for the failure is due to xUnit's cctor or the non-use of GeneratedResolver.
It's not very good.

Reply:
> some of the tests for IL2CPP are failing.

Is this something we can improve on by changing the pipeline to run these tests for every PR/CI?

Reply:
For example, this command.

```
# Build test executable binary
- run: /opt/Unity/Editor/Unity -quit -batchmode -nographics -silent-crashes -logFile -projectPath . -executeMethod UnitTestBuilder.BuildUnitTest /headless /ScriptBackend IL2CPP /BuildTarget StandaloneWindows64

# Run player
- run: ./MessagePack/bin/UnitTest/StandaloneLinux64_IL2CPP/test.exe
```